train.py

```diff
@@ -37,51 +39,74 @@
 X_test = zero_pad(X_test, SEQUENCE_LENGTH)
 
 # Different placeholders
-batch_ph = tf.placeholder(tf.int32, [None, SEQUENCE_LENGTH])
-target_ph = tf.placeholder(tf.float32, [None])
-seq_len_ph = tf.placeholder(tf.int32, [None])
-keep_prob_ph = tf.placeholder(tf.float32)
+with tf.name_scope('inputs'):
+    batch_ph = tf.placeholder(tf.int32, [None, SEQUENCE_LENGTH], name='batch_ph')
+    target_ph = tf.placeholder(tf.float32, [None], name='target_ph')
+    seq_len_ph = tf.placeholder(tf.int32, [None], name='seq_len_ph')
+    keep_prob_ph = tf.placeholder(tf.float32, name='keep_prob_ph')
 
 # Embedding layer
-embeddings_var = tf.Variable(tf.random_uniform([vocabulary_size, EMBEDDING_DIM], -1.0, 1.0), trainable=True)
-batch_embedded = tf.nn.embedding_lookup(embeddings_var, batch_ph)
+with tf.name_scope('Embedding_layer'):
+    embeddings_var = tf.Variable(tf.random_uniform([vocabulary_size, EMBEDDING_DIM], -1.0, 1.0), trainable=True)
+    tf.summary.histogram('embeddings_var', embeddings_var)   
+    batch_embedded = tf.nn.embedding_lookup(embeddings_var, batch_ph)
 
 # (Bi-)RNN layer(-s)
 rnn_outputs, _ = bi_rnn(GRUCell(HIDDEN_SIZE), GRUCell(HIDDEN_SIZE),
                         inputs=batch_embedded, sequence_length=seq_len_ph, dtype=tf.float32)
+tf.summary.histogram('rnn_outputs', rnn_outputs)   
 # rnn_outputs, _ = rnn(GRUCell(hidden_size), inputs=batch_embedded, sequence_length=seq_len_ph, dtype=tf.float32)
 
+
 # Attention layer
-attention_output, alphas = attention(rnn_outputs, ATTENTION_SIZE, return_alphas=True)
+with tf.name_scope('Attention_layer'):
```

Comment:
These histograms show how these parameters convergenced, but you can remove some of it if some seems verbose.